### DISCUSSION TITLE
Custom linear colormaps

### DISCUSSION THREAD

**Discussion by delbronski** [developmentseed/titiler#739] [Q&A] [answered]
Is there a way to provide a custom linear colormap with TiTiler. 

I saw this example in the docs, which allows users to pass a linear definition.

https://developmentseed.org/titiler/examples/code/tiler_with_custom_colormap/

But I was wondering if there was a way that I could do something like this out of the box with TiTiler. I did not see any example in the docs of how something like this could be done. 

For example, let's say I had data that looked like this:

  "values": [
    0.14231166243553162,
    0.3904677703976631,
    0.6386238783597946,
    0.8867799863219261,
    1.1349360942840576
  ],
  "colors": [
    "#2A5C84",
    "#74AE56",
    "#FFE733",
    "#FF8C01",
    "#ED2938"
  ]
  
  And I wanted something like the example below where the values correspond to the relative positions of each color in a linear scale:
  
`cmap = clr.LinearSegmentedColormap.from_list('custom', 
                                             [(0.14231166243553162,    '#2A5C84'),
                                              ( 0.3904677703976631, '"#74AE56").... etc], N=256)`
                                              
Would I need to add my own custom dependency to achieve something like this?

Thanks!

**Comment by vincentsarago** [comment, marked as answer]
yeah you would need to create a custom colormap dependency https://developmentseed.org/titiler/advanced/customization/#custom-colormap

**Reply by delbronski** [reply]
Thanks! I thought so, but just wanted to make sure I was not missing something. 

I don't think it should be that hard.

**Reply by vincentsarago** [reply]
> I don't think it should be that hard.

I hope so, if any trouble please come back and we'll update the docs 🙏
